go.mod

```diff
@@ -2,18 +2,19 @@ module github.com/GoogleContainerTools/kpt
 
 go 1.17
 
+replace github.com/GoogleContainerTools/kpt/porch/api => ./porch/api
```

Comment:
This breaks `go install`: https://kpt.dev/installation/?id=source

```
$ go install -v github.com/GoogleContainerTools/kpt@main
go install: github.com/GoogleContainerTools/kpt@main (in github.com/GoogleContainerTools/kpt@v1.0.0-beta.14.0.20220404215315-97c704191b82):
        The go.mod file for the module providing named packages contains one or
        more replace directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module.
```

Reply:
We have a couple of places in the catalog repo relies on this.

Reply:
too much fun with `go.mod` :) should we revert this PR to unblock PRs on `kpt-function-catalog` repo ?

Reply:
I'd just go back to using hardcoded version for this dependency. should be easy fix. we did it once before.

Reply:
#2967 should resolve the issue.